geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisList.java

```diff
@@ -92,6 +93,19 @@ public int llen() {
     return elementList.size();
   }
 
+  /**
+   * @param region, the region to set on
+   * @param key, the key to set on
+   * @param index, the adjusted index to set (adjusted to a positive index)
+   * @param value, the value to set
+   * @return boolean if set
+   */
+  public boolean lset(Region<RedisKey, RedisData> region, RedisKey key, int index, byte[] value) {
+    byte[] delta = elementList.set(index, value);
+    storeChanges(region, key, new ReplaceByteArrayAtOffset(index, delta));
```

Comment:
You don't want to ship the oldValue returned by "set" as the value in that delta. You can ignore the value returned by "set" (get rid of the "delta" variable) and pass "value" to the ReplaceByteArrayAtOffset constructor.